packages/core/src/drawer/drawer.stories.ts

```diff
@@ -0,0 +1,89 @@
+import type { Drawer } from './drawer.js';
+import type { StoryObj } from '@storybook/web-components';
+import { html } from 'lit';
+import '../button/register.js';
+import './register.js';
+
+export default {
+  title: 'Drawer',
+  args: {
+    attachment: 'right',
+    buttonSize: 'sm'
+  },
+  argTypes: {
+    attachment: {
+      control: 'radio',
+      options: ['right', 'left', 'top', 'bottom']
+    },
+    buttonSize: {
+      control: 'radio',
+      options: ['sm', 'md', 'lg']
+    }
+  }
+};
+
+const onClick = (event: Event & { target: HTMLElement }): void => {
+  (event.target.nextElementSibling as Drawer).showModal();
+};
+
+export const API: StoryObj = {
+  render: ({ attachment, buttonSize }) => html`
+    <sl-button @click=${onClick}>Show Drawer</sl-button>
+    <sl-drawer .attachment=${attachment} .closeButtonSize=${buttonSize}>
```

Comment:
I'm not sure if `closeButtonSize` is a valid public property. Perhaps it should be something internal? So you could have a design token with the size of the close button?

Reply:
Perhaps have a `no-close-button` attribute that causes the component to not render a close button? For example when you have a form inside the drawer that has a "Cancel" button?

Reply:
But perhaps this is beyond the scope of the PoC :)